guides/deployment/fly.md

```diff
@@ -0,0 +1,281 @@
+# Deploying on Fly
+
+## What we'll need
+
+The only thing we'll need for this guide is a working Phoenix application. For those of us who need a simple application to deploy, please follow the [Up and Running guide](up_and_running.html).
+
+## Goals
+
+Our main goal for this guide is to get a Phoenix application running on Fly.io.
+
+## Steps
+
+Let's separate this process into a few steps so we can keep track of where we are.
+
+- Install the Fly CLI
+- Sign up for Fly
+- Make our project ready for Fly
+- Create and set up our Fly application
+- Provision a database
+- Deploy time!
+- Helpful Fly resources
+
+## Installing the Fly CLI
+
+Follow the instructions [here](https://fly.io/docs/getting-started/installing-flyctl/) to install the command-line interface for the Fly platform.
+
+## Sign up for Fly
+
+We can [sign up for an account](https://fly.io/docs/getting-started/login-to-fly/) using the CLI.
+
+```console
+$ fly auth signup
+```
+
+Fly has a [free tier](https://fly.io/docs/about/pricing/) for applications without a database. A credit card is required when setting up an account to help prevent abuse. See the [pricing](https://fly.io/docs/about/pricing/) page for more details.
+
+## Make our project ready for Fly
+
+For this guide, we'll use a Dockerfile and build a release for our Fly deployment. Internally, Fly's networking uses IPv6, so there is a little config we can do to our application to make it a smooth experience.
+
+### Use releases
+
+Configure the application to [Deploy using Releases](releases.html) including the section on Containers. There is a guide for deploying Elixir applications in the [Fly documentation](https://fly.io/docs/getting-started/elixir/) that you can refer to for this as well.
+
+### Runtime configuration
+
+After following the [Deploy using Releases](releases.html) steps with the `config/runtime.exs` file, we are ready to configure it for Fly.
+
+Update the `config/runtime.exs` file to follow this example:
+
+```elixir
+import Config
+
+if config_env() == :prod do
+  secret_key_base =
+    System.get_env("SECRET_KEY_BASE") ||
+      raise """
+      environment variable SECRET_KEY_BASE is missing.
+      You can generate one by calling: mix phx.gen.secret
+      """
+
+  app_name =
+    System.get_env("FLY_APP_NAME") ||
+      raise "FLY_APP_NAME not available"
+
+  config :my_app, MyAppWeb.Endpoint,
+    server: true,
+    url: [host: "#{app_name}.fly.dev", port: 80],
+    http: [
+      port: String.to_integer(System.get_env("PORT") || "4000"),
+      # IMPORTANT: support IPv6 addresses
+      transport_options: [socket_opts: [:inet6]]
```

Comment:
Check config/runtime.exs  in master, there is a more convenient of setting inet6 in there which is by setting the ip address with an ipv6. :)